### DISCUSSION TITLE
Camera names limitation

### DISCUSSION THREAD

**Discussion by dorfil** [bluenviron/mediamtx#1950] [Questions]
Hi!

I've this use case where I would need to add camera names with blank spaces, or japanese letters for example.

Is there any reason why camera names are limited to "...only alphanumeric characters, underscore, dot, tilde, minus or slash"?

I tried disabling the check that is applied to camera names (regex: ^[0-9a-zA-Z_-/.~]+$).

After that change applied, I can add camera names with blank spaces and so on. Accesing the camera over RTSP and HLS worked with no problem, since the URL is encoded, and then decoded by mediamtx.

Do you see that this could cause any problem ?

Thanks!

**Comment by aler9** [comment]
Hello, adding blank spaces is something that would surely break some client/protocol combination. We can't risk, sorry.

[a comment by an automated account is not shown]
